### PR TITLE
[jsk_perception/point_pose_extractor] Use DiagnosticNodelet to improve subscribing

### PR DESCRIPTION
**Pull request by nakane11**
### Problem
- During the fetch's kitchen demo, this node in trashbin_occupancy_detector continues logging `"wait for subscriberes .."`
- If `SetTemplate` service is requested before `imagefeature_cb` is called, `cv::warpPerspective` raises an error because `PinholeCameraModel` has not been set
- `cv::findHomography()` could returns empty matrix
https://docs.opencv.org/3.3.0/d9/d0c/group__calib3d.html#ga4abc2ece9fab9398f2e560d53c8c9780

I modified point_pose_extractor to inherit DiagnosticNodelet thanks to @iory and added some processing to avoid the errors above.

### CODE REVIEW

File: jsk_perception/CMakeLists.txt [hunk 1]
```diff
@@ -234,6 +234,7 @@ jsk_add_nodelet(src/sparse_image_decoder.cpp "jsk_perception/SparseImageDecoder"
 jsk_add_nodelet(src/color_histogram.cpp "jsk_perception/ColorHistogram" "color_histogram")
 jsk_add_nodelet(src/background_substraction_nodelet.cpp "jsk_perception/BackgroundSubstraction" "background_substraction")
 jsk_add_nodelet(src/fisheye_to_panorama.cpp "jsk_perception/FisheyeToPanorama" "fisheye_to_panorama")
+jsk_add_nodelet(src/point_pose_extractor.cpp "jsk_perception/PointPoseExtractor" "point_pose_extractor")
```
Comment (iory) [inline jsk_perception/CMakeLists.txt:237]: Could you add test?
https://github.com/jsk-ros-pkg/jsk_recognition/blob/0de1e9ac4ec9ea27c548d835c9ca5e94a6101808/jsk_perception/CMakeLists.txt#L468-L582

**Review by iory — CHANGES_REQUESTED** [review]

**Comment by nakane11** [comment]
I replaced images for sample because previously object dtection did not succeed.
It enabled point_pose_extractor.test to check if object detection is published.

Should I split these commits?

File: jsk_perception/CMakeLists.txt (hunk 1 above)
Comment (nakane11) [reply jsk_perception/CMakeLists.txt:237]: I added the test in 4a7aeda

Comment (iory) [reply jsk_perception/CMakeLists.txt:237]: Nice!

**Comment by iory** [comment]
> Should I split these commits?

You don't have to.

Could you update the document for other users?
https://github.com/jsk-ros-pkg/jsk_recognition/blob/master/doc/jsk_perception/nodes/point_pose_extractor.md
For example, how to use the samples you made, etc.

**Comment by nakane11** [comment]
Sorry for the late update. 
I edited the document.

File: doc/jsk_perception/nodes/point_pose_extractor.md [hunk 2]
```diff
@@ -8,7 +8,8 @@
 
 - `/ImageFeature0D` (`posedetection_msgs::ImageFeature0D`)
 
-  Image, camera and tempalte feature information. 
+  Image, camera and tempalte feature information.
```
Comment (Affonso-Gui) [inline doc/jsk_perception/nodes/point_pose_extractor.md:11]: template

Comment (nakane11) [reply doc/jsk_perception/nodes/point_pose_extractor.md]: Thank you. I fixed it.

**Comment by nakane11** [comment]
Template added from mouse callback is saved as `filename`. 
https://github.com/jsk-ros-pkg/jsk_recognition/blob/1704f9b974cf9901e8feb64639c32267e8bde94c/jsk_perception/include/jsk_perception/point_pose_extractor.h#L659-L667

1. `fname.stem()` and `fname.extension()` put string between " ", so when I input `hoge.png`, `ss` contained `"hoge"_wrap".png"`. I modified it to contain `hoge_wrap.png`.
2. Previously pictures are saved in .ros directry because `fname.stem()` returns filename without path. I attatched `fname.parent_path()` to stem. Now we can save a template as `/tmp/hoge.png`.

**Review by iory — APPROVED** [review]

File: jsk_perception/plugins/nodelet/libjsk_perception.xml [hunk 3]
```diff
@@ -290,6 +290,9 @@
          type="jsk_perception::VirtualCameraMono"
          base_class_type="nodelet::Nodelet">
   </class>
+  <class name="jsk_perception/PointPoseExtractor"
+         type="jsk_perception::PointPoseExtractor"
+         base_class_type="nodelet::Nodelet">
```
Comment (k-okada) [inline jsk_perception/plugins/nodelet/libjsk_perception.xml:295]: @nakane11 are we missing `</class>` tag?

Comment (nakane11) [reply jsk_perception/plugins/nodelet/libjsk_perception.xml:295]: I'm sorry, I overlooked that. 
19d2e95 added the end tag.

**Pull request merged**